File: web/src/types/lookup.ts
export interface LookupListItem {
  id: number;
  query_text: string;
  query_language: string | null;
  matched_hs_code: string | null;
  matched_description_vn: string | null;
  confidence_score: number | null;
  search_method: string;
  is_verified: boolean;
  created_at: string;
}

export interface PaginatedLookupListResponse {
  items: LookupListItem[];
  total: number;
  limit: number;
  offset: number;
}

export interface LookupDetailHSCode {
  code: string;
  description_vn: string | null;
  description_en: string | null;
  duty_rate: string | null;
  vat_rate: string | null;
}

export interface LookupDetail {
  id: number;
  query_text: string;
  query_language: string | null;
  matched_hs_code: LookupDetailHSCode | null;
  correct_hs_code: LookupDetailHSCode | null;
  classification_data: { material: string; function: string } | null;
  practical_notes: string[] | null;
  process_logs:
    | {
        step: string;
        status: string;
        message: string;
        duration_ms: number;
        details?: Record<string, unknown>;
      }[]
    | null;
  nlm_raw_response: string | null;
  confidence_score: number | null;
  search_method: string;
  is_verified: boolean;
  verified_at: string | null;
  notes: string | null;
  created_at: string;
}

File: api/app/schemas/lookup.py
"""Schemas for lookup history list and detail responses."""

from typing import Any

from pydantic import BaseModel


class LookupListItem(BaseModel):
    """Single lookup record in the list response."""

    id: int
    query_text: str
    query_language: str | None
    matched_hs_code: str | None
    matched_description_vn: str | None
    confidence_score: float | None
    search_method: str
    is_verified: bool
    created_at: str


class PaginatedLookupListResponse(BaseModel):
    """Paginated list of lookup records."""

    items: list[LookupListItem]
    total: int
    limit: int
    offset: int


class LookupDetailHSCode(BaseModel):
    """HS code details for lookup detail response."""

    code: str
    description_vn: str | None
    description_en: str | None
    duty_rate: str | None
    vat_rate: str | None


class LookupDetailResponse(BaseModel):
    """Full lookup record detail response."""

    id: int
    query_text: str
    query_language: str | None
    matched_hs_code: LookupDetailHSCode | None
    correct_hs_code: LookupDetailHSCode | None
    classification_data: dict[str, Any] | None
    practical_notes: list[str] | None
    process_logs: list[dict[str, Any]] | None
    nlm_raw_response: str | None = None
    confidence_score: float | None
    search_method: str
    is_verified: bool
    verified_at: str | None
    notes: str | None
    created_at: str
